op-plasma/daserver.go

```diff
@@ -126,26 +132,44 @@ func (d *DAServer) HandlePut(w http.ResponseWriter, r *http.Request) {
 
 	input, err := io.ReadAll(r.Body)
 	if err != nil {
+		d.log.Error("Failed to read request body", "err", err)
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	key := path.Base(r.URL.Path)
-	comm, err := hexutil.Decode(key)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		return
-	}
+	if r.URL.Path == "/put" || r.URL.Path == "/put/" { // without commitment
 
-	if err := d.store.Put(r.Context(), comm, input); err != nil {
-		d.log.Info("Failed to store commitment to the DA server", "err", err, "key", key)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-	if _, err := w.Write(comm); err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		return
+		comm := GenericCommitment(crypto.Keccak256Hash(input).Bytes())
```

Comment:
why are we still generating a commitment in this case ? is this like a placeholder for now.

Reply:
this case is indicating that the commitment should be generated by the test DA server since its not receiving any commitment value in the URL path which we do to ensure compliance with the existing interactions stipulated via [spec](https://specs.optimism.io/experimental/plasma.html#da-server). We could extend this test server to use the `DA_SERVICE` feature for more explicit control flow deduction. Not super opinionated and will defer to @trianglesphere  🙏🏻

Reply:
This is mainly a place holder / implementing the server side computation for keccak.  This implementation works well for our use cases